src/apps/inverse-firm/ethereum/inverse-firm.loan.contract-position-fetcher.ts

```diff
@@ -108,4 +113,80 @@ export class EthereumInverseFirmLoanContractPositionFetcher extends ContractPosi
 
     return [supplied, borrowed];
   }
+
+  async getRewardTokenBalances({
+    address,
+    multicall,
+    contract,
+  }: GetTokenDefinitionsParams<SimpleMarket>): Promise<BigNumberish[]> {
+    const personalEscrow = await contract.escrows(address);
+    if (personalEscrow === ZERO_ADDRESS) return [];
+
+    const escrowContract = this.contractFactory.rewardableEscrow({
+      address: personalEscrow.toLowerCase(),
+      network: this.network,
+    });
+
+    let dbrClaimable: BigNumberish = 0;
+
+    try {
+      // escrow may or not be with the rewards feature
+      dbrClaimable = await multicall.wrap(escrowContract).claimable();
+    } catch (e) {
+      // no claimable rewards feature for this market
+    }
+
+    return [
+      dbrClaimable,
+    ];
+  }
+
+  async getRewardTokenAddresses() {
+    return [
+      this.dbrAddress,
+    ];
+  }
+
+  async getRewardRates({
+    multicall
+  }) {
+    const dbrDistributorContract = this.contractFactory.dbrDistributor({
+      address: this.dbrDistributor,
+      network: this.network,
+    });
+    const xinvContract = this.contractFactory.xinv({
+      address: this.xinvAddress,
+      network: this.network,
+    });
+
+    const [
+      rewardRate,
+      dbrDistributorSupply,
+      xinvExRate,
+    ] = await Promise.all([
+      multicall.wrap(dbrDistributorContract).rewardRate(),
+      multicall.wrap(dbrDistributorContract).totalSupply(),
+      multicall.wrap(xinvContract).exchangeRateStored(),
+    ]);
+
+    const [dbrtoken, invToken] = await Promise.all([
+      this.appToolkit.getBaseTokenPrice({
+        network: this.network,
+        address: this.dbrAddress,
+      }),
+      this.appToolkit.getBaseTokenPrice({
+        network: this.network,
+        address: this.invAddress,
+      }),
+    ]);
+
+    if (!dbrtoken || !invToken) return [0];
+
+    const invStakedViaDistributor = xinvExRate.mul(dbrDistributorSupply);
+    const dbrYearlyRewardRate = rewardRate.mul(31536000);
+    const dbrInvExRate = dbrtoken.price / invToken.price;
```

Comment:
I don't know how accurate you want the APR to be, but calling price from oracle will most likely result in a more precise number matching the one on your FIRM's UI 
(I'm guessing the app has oracles for base token prices right ?)

Reply:
> I don't know how accurate you want the APR to be, but calling price from oracle will most likely result in a more precise number matching the one on your FIRM's UI (I'm guessing the app has oracles for base token prices right ?)

No we use coingecko prices for the DBR APR on the UI, DBR has no oracle price, it's not a collateral or a borrowable asset.